client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -165,6 +190,68 @@ export class Mentions extends React.Component {
 		editor.getBody().focus();
 	}
 
+	onKeyDown = ( e ) => {
+		const { keyCode } = e;
+
+		if ( includes( [ VK.DOWN, VK.UP ], keyCode ) ) {
+			const selectedIndex = this.getSelectedSuggestionIndex();
+			let nextIndex;
+
+			if ( selectedIndex !== null ) {
+				if ( keyCode === VK.DOWN ) {
+					nextIndex = selectedIndex + 1;
+
+					if ( nextIndex === this.matchingSuggestions.length ) {
+						nextIndex = 0;
+					}
+				} else {
+					nextIndex = selectedIndex - 1;
+
+					if ( nextIndex < 0 ) {
+						nextIndex = this.matchingSuggestions.length - 1;
+					}
+				}
```

Comment:
is this supposed to update the selection on any key press which is not `VK.DOWN`? It feels like it should only match on `VK.UP` and if so then being a bit more explicit would clean it up considerably.

incidentally, if the operation we are actually wanting to do is simply scroll in the direction specified by these keys, then we can simplify taking in to mind the modulus operation I left in the previous comment

```js
const delta = get( {
	[ VK.DOWN ]: 1,
	[ VK.UP ]: -1,
}, keyCode, 0 );

const nextIndex = ( selectedIndex + delta ) % this.matchingSuggestions.length;
```

in these few lines we are able to replicate the full nested `if` structure (and the early/late abort for non-matching keys thanks to the default mapping of `0` inside `get`) in two short declarative lines :smile: